checks/apps/namd/namd_check.py

```diff
@@ -1,133 +1,296 @@
+            prepare_cmd += [f'cd {self.srcdir}']
+
+        if self.builddir:
+            prepare_cmd += [f'mkdir -p {self.builddir}', f'cd {self.builddir}']
+
+        if self.builddir:
+            configure_cmd = [
+                os.path.join(
+                    os.path.relpath(self.configuredir, self.builddir), 'config'
+                )
+            ]
         else:
-            if self.current_system.name in ['eiger', 'pilatus']:
-                self.num_tasks = 2048
-                self.num_tasks_per_node = 128
-            else:
-                self.num_tasks = 16
-                self.num_tasks_per_node = 1
+            configure_cmd = [os.path.join(self.configuredir, 'config')]
+
+        if self.config_opts:
+            configure_cmd += self.config_opts
+
+        config_dir = 'Linux-ARM64-g++.cuda'
+
+        make_cmd = ['make -j']
+        if self.max_concurrency is not None:
+            make_cmd += [str(self.max_concurrency)]
+
+        if self.make_opts:
+            make_cmd += self.make_opts
+
+        return prepare_cmd + [
+            ' '.join(configure_cmd),  # Run ./config
+            f'cd {config_dir}',  # cd to directory created by .config
+            ' '.join(make_cmd),  # make
+        ]
+
+
+@rfm.simple_test
+class NamdBuildTest(rfm.CompileOnlyRegressionTest):
+    '''
+    Test NAMD build from source.
+    '''
+
+    descr = 'NAMD Build Test'
+    valid_prog_environs = ['+namd-single-node-dev']
+    valid_systems = ['*']
+    build_system = AutotoolsCustom()
+    sourcesdir = None
+    maintainers = ['SSA']
+    namd_sources = fixture(namd_download, scope='session')
+    build_locally = False
+    tags = {'uenv'}
 
     @run_before('compile')
     def prepare_build(self):
-        # Reset sources dir relative to the SCS apps prefix
-        self.sourcesdir = os.path.join(self.current_system.resourcesdir,
-                                       'NAMD', 'prod')
+        self.uarch = uenv.uarch(self.current_partition)
+        self.build_system.builddir = self.stagedir
+        self.skip_if_no_procinfo()
+        cpu = self.current_partition.processor
+        self.build_system.max_concurrency = cpu.info['num_cpus_per_socket']
+
+        tarsource = os.path.join(
+            self.namd_sources.stagedir,
+            f'NAMD_{self.namd_sources.version}_Source.tar.gz',
+        )
+
+        # Extract source code and compiler bundled Charm++
+        self.prebuild_cmds = [
+            f'tar --strip-components=1 -xzf {tarsource} -C {self.stagedir}',
+            'tar -xf charm-8.0.0.tar',  # INFO: Depends on the NAMD version
+            'cd charm-8.0.0',
+            f'./build charm++ multicore-linux-arm8 gcc --with-production --enable-tracing -j {self.build_system.max_concurrency}',
+            'cd ..',
+            # Link against tcl8.6 (provided by the UENV)
+            'sed -i \'s/-ltcl8.5/-ltcl8.6/g\' arch/Linux-ARM64.tcl',
+        ]
+
+        prefix = '/user-environment/env/develop-single-node'
```

Comment:
try to use os.getenv('UENV_MOUNT_POINT') ?

Reply:
Isn't that just `/user-environment`? This needs to point to the view. It also returns `None` at this stage. Can I get the view programmatically from `reframe.yaml`?